src/test/java/emissary/core/channels/LoggingInputStreamFactoryTest.java

```diff
@@ -0,0 +1,127 @@
+package emissary.core.channels;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.core.read.ListAppender;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class LoggingInputStreamFactoryTest {
+    private ListAppender<ILoggingEvent> appender = null;
+    private final Logger logger = (Logger) LoggerFactory.getLogger(LoggingInputStreamFactoryTest.class);
+
+    @BeforeEach
+    void setup() throws IOException {
+        appender = new ListAppender<>();
+        appender.start();
+        logger.addAppender(appender);
+    }
+
+    @AfterEach
+    void detach() throws Exception {
+        logger.detachAppender(appender);
+    }
+
+    @Test
+    void testChannelFactory() throws IOException {
+        final byte[] bytes = new byte[100];
+        final TestInputStreamFactory tisf = new TestInputStreamFactory(bytes);
+        final InputStreamFactory loggingFalseIsf = LoggingInputStreamFactory.create(tisf, "Identifier", logger, false);
+
+        testLogging(appender, loggingFalseIsf, false);
+
+        final InputStreamFactory loggingTrueIsf = LoggingInputStreamFactory.create(tisf, "Identifier", logger, true);
+
+        testLogging(appender, loggingTrueIsf, true);
+    }
+
+    private static void testLogging(final ListAppender<ILoggingEvent> appender, final InputStreamFactory isf,
```

Comment:
1. Should add a `@SuppressWarnings("ResultOfMethodCallIgnored")` annotation to this method to suppress compiler warnings.
2. Suggest removing `static` modifier and `appender` parameter.

Reply:
1. Now checks method return values.
2. Changed.